src/plugins/intel_gna/src/gna_plugin.cpp

```diff
@@ -201,7 +201,31 @@ void GNAPlugin::copyInputData(T *dst,
     }
 }
 
-void GNAPlugin::ExportScores(void *ptr_dst,
+template <typename T>
+void GNAPlugin::UpscaleAndCast(T* dst_ptr, const uint8_t* input_ptr, const Precision& precision_in, const float& scale_factor) {
+    switch (precision_in) {
```

Comment:
we should avoid switch/is etc in this perf-cruitical function
we should move it outside a loop.